### ISSUE TITLE
Remove mention of Heroku's deprecated free tier in heroku-server/README.md

### ISSUE THREAD

**Issue by AntonioClarke** [GoogleChrome/lighthouse-ci#869]
The documentation for setting up a [custom lighthouse server with Heroku](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/recipes/heroku-server/README.md) specifically call out that it has a free tier which fulfills the requirements for a custom lighthouse server. 

As of [November 2022](https://help.heroku.com/RSBRUH58/removal-of-heroku-free-product-plans-faq), this is no longer the case. I only realized this after I'd finished creating an account, verifying my email, installing an Authenticator app on my cell phone, authenticating with Heroku, and then trying to set up an app but being blocked by needing to add a payment method... 

By updating the docs to avoid the misleading mention of a free tier, other devs time might not be wasted like mine was.

**Comment by paulirish** [comment]
Good idea. Do you want to propose the wording changes in a pull request?
